src/Orleans.Core/Runtime/OutsideRuntimeClient.cs

```diff
@@ -110,6 +117,7 @@ public IStreamProviderRuntime CurrentStreamProviderRuntime
             this.clientMessagingOptions = clientMessagingOptions.Value;
             this.typeMapRefreshInterval = typeManagementOptions.Value.TypeMapRefreshInterval;
             this.responseTimeout = clientMessagingOptions.Value.ResponseTimeout;
+            this.collectThreadTimeTrackingStats = statisticsOptions.Value.CollectionLevel >= StatisticsLevel.Verbose2;
```

Comment:
this variable shouldn't be on the OutsideRuntimeClient level, it should be with the StatisticGroup which collects `ThreadTimeTrackingStats`